Makes simple search request

webinista opened the pull request:
WIP for department-of-veterans-affairs/messaging-fe#119
- Added SET_SEARCH, SEARCH_SUCCESS, SEARCH_FAILURE constants.
- Changed messaging search input to use ErrorableTextInput.
- Passes current folder ID to MessageSearch component.
- Added SET_SEARCH constant
- Adding an onSearchChange handler.
- Updating search state structure to be {params, advanced}
- Updating PropTypes definitions for MessageSearch, MessageSearchAdvanced

Related to department-of-veterans-affairs/messaging-fe#93

webinista committed:
Makes simple search request

WIP for department-of-veterans-affairs/messaging-fe#119

- Added SET_SEARCH, SEARCH_SUCCESS, SEARCH_FAILURE constants.
- Changed messaging search input to use ErrorableTextInput.
- Passes current folder ID to MessageSearch component.
- Added SET_SEARCH constant
- Adding an onSearchChange handler.
- Updating search state structure to be {params, advanced}
- Updating PropTypes definitions for MessageSearch, MessageSearchAdvanced

Related to department-of-veterans-affairs/messaging-fe#93

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#119-part1

webinista commented:
To be clear: this just builds the query URL. It's a work in progress, but I wanted to get feedback on what's here now.

U-DON commented on src/js/messaging/components/MessageSearch.jsx:
```diff
@@ -25,23 +60,41 @@ class MessageSearch extends React.Component {
+      value: React.PropTypes.string,
+      dirty: React.PropTypes.bool
+    })
+  })
```
Add these prop types to `MessageSearchAdvanced` too.

U-DON commented on src/js/messaging/containers/Folder.jsx:
```diff
@@ -139,6 +141,7 @@ export class Folder extends React.Component {
 
   render() {
     const folderName = _.get(this.props.attributes, 'name');
+    const folderId = _.get(this.props.params, 'id', 0);
```
The id in the URL corresponds to the folder being requested, and the one in attributes corresponds to the id of the actual current folder. This can get out of sync when the folder fails to load. It might be more intuitive to use the attributes id since that's the folder that's currently displayed.

U-DON commented on src/js/messaging/reducers/search.js:
```diff
@@ -1,30 +1,47 @@
+    case SET_ADVSEARCH_START_DATE:
+      return set('params.dateRange.start', action.date, state);
+    case SET_SEARCH_PARAM:
+      return set(action.path, action.field, state);
```
Since it's always only affecting `params`, maybe set it on `params.${action.path}` and cut the `params.` prefix from the path arguments when calling the action creator.

U-DON reviewed (commented):
LGTM with some minor comments.

webinista committed:
Updating props, search params path, removing action / reducer

- Added onFieldChange, params props to MessageSearchAdvanced.
- Made params prop required in MessageSearch, MessageSearchAdvanced.
- Changing this.props.params to this.props.attributes.
- Removed SEARCH_SUCCESS action / reducer (to be resolved in a follow up PR)
- Modified object path of for search results.

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#119-part1

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#119-part1
